### ISSUE TITLE
Using / Decoding Service Account API Tokens results in invalid base64 error

### ISSUE THREAD

**Issue by illode** [kanidm/kanidm#2631]
### I did this
I was trying to configure https://github.com/owncloud/ocis to use the Kanidm LDAP server, but it was never able to authenticate. The Kanidm logs would print
`[info]: Unable to decode token | event_tag_id: 10 | e: InvalidBase64`

I did some more testing:
```bash
kanidm service-account api-token generate ocis-service-account "OCIS"
echo -n '<Very Long Token>' | base64 -d | jq
```
it at least partially decodes, but base64 complains:
```json
base64: invalid input
{
  "alg": "ES256",
  "jwk": {
    "kty": "EC",
    "crv": "P-256",
    "x": "1GRfPJ7Qkc6yxxilWj8thidn_Wqpgqq54vP5BwhmLdA",
    "y": "LO7KzYBIKaSNOQXrZymeZzoiMpqdDOliJpmKyDuFABM",
    "alg": "ES256",
    "use": "sig",
    "kid": "4c3fd4e9d2fb3ab18f41fec3d088d4e5ec8665561fa1e35c7d9be8a9eb2f98b3"
  },
  "kid": "4c3fd4e9d2fb3ab18f41fec3d088d4e5ec8665561fa1e35c7d9be8a9eb2f98b3"
}
```
The most inexplicable thing, though, is that using `ldapsearch`, this works:
```bash
LDAPTLS_REQCERT=never ldapsearch -H ldaps://10.0.32.21:3636 -D 'dn=token' -x -w "<Very Long Token>" -b 'dc=kanidm,dc=example,dc=com' '*'
# success
```
but this fails:
```bash
TESTOKEN='<Very Long Token>'
LDAPTLS_REQCERT=never ldapsearch -H ldaps://10.0.32.21:3636 -D 'dn=token' -x -w "$TESTOKEN" -b 'dc=kanidm,dc=example,dc=com' '*'
# ldap_bind: Other (e.g., implementation specific) error (80)
#    additional info: NotAuthenticated
```
Somehow `ldapsearch` when the token is used in-line, but fails when it's an environment variable.

After testing using some more tokens, I found that some of them gave a different error:
`[info]: Unable to decode token | event_tag_id: 10 | e: InvalidHeaderFormat`

### Kanidm version details

- Output of `kanidm(d) version`:
`kanidmd 1.1.0-rc.16`
- Are you running it in a container? If so, which image/tag?:
No
- If not a container, how'd you install it:
NixOS / Nixpkgs
- Operating System / Version (On Unix please post the output of `uname -a`):
`Linux auth 6.1.79 #1-NixOS SMP PREEMPT_DYNAMIC Fri Feb 23 08:12:53 UTC 2024 x86_64 GNU/Linux`

### Any other comments

As an aside, when I tried to copy the version, I found that both `kanidm` and `kanidmd` panic if they try to output to a broken pipe:
```bash
kanidmd version | qwe
qwe: command not found
thread 'main' panicked at library/std/src/io/stdio.rs:1021:9:
failed printing to stdout: Broken pipe (os error 32)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by yaleman** [comment]
JWTs are three base64'd elements combined with dots, so you'll need to split them to decode them with `base64` on the CLI.

The fact it's failing in ldapsearch on the command line is... weird, but that more like something to do with your CLI than a Kanidm-related issue?

The printing to a broken pipe thing seems to be a well-known issue with rust: https://github.com/rust-lang/rust/issues/46016

**Comment by illode** [comment]
> JWTs are three base64'd elements combined with dots, so you'll need to split them to decode them with `base64` on the CLI.

I completely glossed over the dots, but even when I split it `base64` complains.

> The fact it's failing in ldapsearch on the command line is... weird, but that more like something to do with your CLI than a Kanidm-related issue?

That's what I thought, so I tried again using bash, which didn't help. Then I tried using a different terminal editor, which didn't help either. After you said this, I tried using a different machine, and it did work, and I just realized I'm an idiot and forgot to put the second T in `TESTTOKEN` when I ran the `ldapsearch` :facepalm: 

That doesn't explain why OCIS isn't working, but now I'm second guessing whether I set it up right.
